src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,105 @@
-import React from 'react';
 import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../react-app-env';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  commentsList, setCommentsList,
+}) => {
```

Comment:
you don't need these props here, because it's bad practice to modify local state values _(commentsList)_.
The best way would be to relate on api:
1. create the post (by using postComment function)
2. use `updateComments` function where you will get comments and update the list _(written in suggestion in `PostDetails`)_
```suggestion
export const NewCommentForm: React.FC<Props> = ({ updateComments }) => {
```